test/applications/cmd/icmp-responder-dns-nse/main.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/connectioncontext"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/connection"
+	"github.com/networkservicemesh/networkservicemesh/pkg/tools"
+	"github.com/networkservicemesh/networkservicemesh/sdk/common"
+	"github.com/networkservicemesh/networkservicemesh/sdk/endpoint"
+	"github.com/sirupsen/logrus"
+	"strings"
+)
+
+var version string
+
+const myCoreDnsPort = 53
+
+func main() {
+	logrus.Info("Starting icmp-responder-dns-nse...")
+	logrus.Infof("Version: %v", version)
+
+	// Capture signals to cleanup before exiting
+	c := tools.NewOSSignalChannel()
+	ipamEndpoint := endpoint.NewIpamEndpoint(nil)
+	routeAddr := makeRouteMutator([]string{"8.8.8.8/30"})
+	if common.IsIPv6(ipamEndpoint.PrefixPool.GetPrefixes()[0]) {
+		routeAddr = makeRouteMutator([]string{"2001:4860:4860::8888/126"})
+	}
+
+	composite := endpoint.NewCompositeEndpointBuilder().
+		Append(
```

Comment:
That actually looks good.

Reply:
Should we better consider to have Build() accept a list of endpoints?
```go
endpoint.NewCompositeEndpoint().Build(endpoint.NewMonitortEndpoint(nil), ..., ipadEndpoint, endpoint.NewConnectionEndpoint(nil))
```
Or even:
```go
endpoint.BuildCompositeEndpoint(endpoint.NewMonitortEndpoint(nil), ..., ipadEndpoint, endpoint.NewConnectionEndpoint(nil))
```

@nickolaev Thoughts?